Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -135,6 +136,12 @@ class TopSitesPanel: UIViewController {
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
         collection?.userInteractionEnabled = false
 
+        if site is SuggestedSite {
+            var deletedSuggestedSites = profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String]
+            deletedSuggestedSites.append(site.url)
+            profile.prefs.setObject(deletedSuggestedSites, forKey: "topSites.deletedSuggestedSites")
+        }
+
```

Comment:
I think the only reason the call below to `removeSiteFromTopSites` works and everything seems okay is that we are essentially doing a no-op since a suggested site doesn't actually exist in the database. I think we should clean this up so we're not always calling `removeSiteFromTopSites` when we don't need to. One way to clean this up could be do bundle up the suggested site deletion into it's own method that adheres to the deferred stuff such as:
```swift
private func deleteTileForSuggestedSite(site: SuggestedSite) -> Success {
    var deletedSuggestedSites = profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String]
    deletedSuggestedSites.append(site.url)
    profile.prefs.setObject(deletedSuggestedSites, forKey: "topSites.deletedSuggestedSites")
   return succeed()
}
```

Then below,
```swift
let deletion = site is SuggestedSite ? deleteTileForSuggestedSite(site) :  profile.history.removeSiteFromTopSites(site)
let newSites = deletion >> {
    self.profile.history.getTopSitesWithLimit(self.maxFrecencyLimit)
}